lib/scout_apm/auto_instrument/rails.rb

```diff
@@ -0,0 +1,73 @@
+
+# In order for this to work, you must add `gem 'parser'` to your Gemfile.
+require 'parser/current'
+
+module ScoutApm
+  module AutoInstrument
+    module Rails
+      def self.rewrite(path, code = nil)
+        code ||= File.read(path)
+
+        ast = Parser::CurrentRuby.parse(code)
+
+        # pp ast
+
+        buffer = Parser::Source::Buffer.new(path)
+        buffer.source = code
+
+        rewriter = Rewriter.new
+
+        # Rewrite the AST, returns a String with the new form.
+        rewriter.rewrite(buffer, ast)
+      end
+
+      class Rewriter < ::Parser::TreeRewriter
+        def initialize
+          super
+
+          @nesting = []
+          @method = []
+        end
+
+        def parent_type?(type, up = 1)
+          parent = @nesting[@nesting.size - up - 1] and parent.type == type
+        end
+
+        # def on_block(node)
+        #   line = node.location.line || 'line?'
+        #   column = node.location.column || 'column?'
+        #   method_name = node.children[0].children[1] || '*unknown*'
+        #
+        #   wrap(node.location.expression, "::ScoutApm::Instruments::AutoInstruments.dynamic_layer('#{method_name}:l#{line}:c#{column}'){", "}")
+        # end
+
+        def on_send(node)
+          return if parent_type?(:block) or @method.empty?
+
+          line = node.location.line || 'line?'
+          column = node.location.column || 'column?'
+          method_name = node.children[1] || '*unknown*'
+
+          wrap(node.location.expression, "::ScoutApm::Instruments::AutoInstruments.dynamic_layer('#{method_name}:l#{line}:c#{column}'){", "}")
```

Comment:
`::ScoutApm::Instruments::AutoInstruments.dynamic_layer` si so... ugly 😜 Thinking about how we can make that better. We mentioned about the invisible space or hot dog. Need to decide on options here

Reply:
I implemented this change in recent commit.